TESTS/mbed_hal/sleep_manager/main.cpp

```diff
@@ -175,53 +213,88 @@ void test_sleep_auto()
-    // * sleep_manager_can_deep_sleep() returns false,
-    // * sleep_manager_can_deep_sleep_test_check() returns true with a 1 ms delay,
-    // * sleep_manager_can_deep_sleep() returns true when checked again.
-    lp_timer.reset();
-    lp_timeout.attach_us(mbed::callback(sleep_manager_unlock_deep_sleep_internal),
-                         DEEP_SLEEP_TEST_CHECK_WAIT_US / 2);
-    lp_timer.start();
-    TEST_ASSERT_FALSE(sleep_manager_can_deep_sleep());
-    TEST_ASSERT_TRUE(sleep_manager_can_deep_sleep_test_check());
-    lp_timer.stop();
-    TEST_ASSERT_UINT64_WITHIN(DEEP_SLEEP_TEST_CHECK_WAIT_DELTA_US, DEEP_SLEEP_TEST_CHECK_WAIT_US / 2,
-                              lp_timer.read_high_resolution_us());
-    TEST_ASSERT_TRUE(sleep_manager_can_deep_sleep());
+        const ticker_info_t *p_ticker_info = get_lp_ticker_data()->interface->get_info();
+
+        // Use LowPowerTimer instead of Timer to prevent deep sleep lock.
+        us_timestamp_t exec_time_unlocked, exec_time_locked;
+        uint32_t start, stop;
+
+        // Deep sleep unlocked:
+        // * sleep_manager_can_deep_sleep() returns true,
+        // * sleep_manager_can_deep_sleep_test_check() returns true instantly.
+        TEST_ASSERT_TRUE(sleep_manager_can_deep_sleep());
+        start = lp_ticker_read();
+        TEST_ASSERT_TRUE(sleep_manager_can_deep_sleep_test_check());
+        stop = lp_ticker_read();
+        exec_time_unlocked = diff_us(start, stop, p_ticker_info);
+
+        // Deep sleep locked:
+        // * sleep_manager_can_deep_sleep() returns false,
+        // * sleep_manager_can_deep_sleep_test_check() returns false with 2 ms delay.
+        sleep_manager_lock_deep_sleep();
+        TEST_ASSERT_FALSE(sleep_manager_can_deep_sleep());
+        start = lp_ticker_read();
+        TEST_ASSERT_FALSE(sleep_manager_can_deep_sleep_test_check());
+        stop = lp_ticker_read();
+        exec_time_locked = diff_us(start, stop, p_ticker_info);
+        TEST_ASSERT_UINT64_WITHIN(DEEP_SLEEP_TEST_CHECK_WAIT_DELTA_US, DEEP_SLEEP_TEST_CHECK_WAIT_US,
+                                  exec_time_locked - exec_time_unlocked);
+
+        // Deep sleep unlocked with a 1 ms delay:
+        // * sleep_manager_can_deep_sleep() returns false,
+        // * sleep_manager_can_deep_sleep_test_check() returns true with a 1 ms delay,
+        // * sleep_manager_can_deep_sleep() returns true when checked again.
+        unlock_deep_sleep = true;
+        lp_ticker_set_interrupt(lp_ticker_read() + us_to_ticks(DEEP_SLEEP_TEST_CHECK_WAIT_US / 2, p_ticker_info));
+
+        start = lp_ticker_read();
+        // Extra wait after setting interrupt to handle CMPOK
+        wait_ns(100000);
+        TEST_ASSERT_FALSE(sleep_manager_can_deep_sleep());
+        TEST_ASSERT_TRUE(sleep_manager_can_deep_sleep_test_check());
+        stop = lp_ticker_read();
+        TEST_ASSERT_UINT64_WITHIN(DEEP_SLEEP_TEST_CHECK_WAIT_DELTA_US, DEEP_SLEEP_TEST_CHECK_WAIT_US / 2,
```

Comment:
```suggestion
        TEST_ASSERT(diff_us(start, stop, p_ticker_info) < DEEP_SLEEP_TEST_CHECK_WAIT_US);
```
With the `DEEP_SLEEP_TEST_CHECK_WAIT_DELTA_US` increased, we now have `TEST_ASSERT_UINT64_WITHIN(delta=1000, expected=1000, actual=1000)` so this assertion needs to be changed.

Reply:
Could even be somthing like that, don't you think?
```
        TEST_ASSERT(diff_us(start, stop, p_ticker_info) >= DEEP_SLEEP_TEST_CHECK_WAIT_US/2);
        TEST_ASSERT(diff_us(start, stop, p_ticker_info) < DEEP_SLEEP_TEST_CHECK_WAIT_US);
```

Reply:
I updated PR as proposed